### PR TITLE
Replace cron with cron parser

### PR DESCRIPTION
**Pull request by Uzlopak**
This is the pendant to https://github.com/agenda/agenda/pull/1150

Switching from [ncb000gt/node-cron](https://www.npmjs.com/package/cron) to [harrisiirak/cron-parser](https://www.npmjs.com/package/cron-parser) for cron-pattern parsing.

    Previously month was 0-based (0=January). Going forward standard Unix pattern is used, which is 1-based (1=January).

    Please update existing cron-patterns that specify a month (4th position of a pattern). The month is now 1 - 12

    1 = January

    2 = February

    3...

    | Example | Execute on 1st of January |
    |---------|---------------------------|
    | Old     | 0 0 1 **0** *             |
    | New     | 0 0 1 **1** *             |

    old Cron patterns

    ```
    * * * * * *
    | | | | | |
    | | | | | +-- Year              (range: 1900-3000)
    | | | | +---- Day of the Week   (range: 1-7, 1 standing for Monday)
    | | | +------ Month of the Year (range: 0-11) NOTE: Difference here
    | | +-------- Day of the Month  (range: 1-31)
    | +---------- Hour              (range: 0-23)
    +------------ Minute            (range: 0-59)
    ```

    new cron patterns

    ```
    * * * * * *
    | | | | | |
    | | | | | +-- Year              (range: 1900-3000)
    | | | | +---- Day of the Week   (range: 1-7, 1 standing for Monday)
    | | | +------ Month of the Year (range: 1-12) NOTE: Difference here
    | | +-------- Day of the Month  (range: 1-31)
    | +---------- Hour              (range: 0-23)
    +------------ Minute            (range: 0-59)
    ```

### CODE REVIEW

**Comment by Uzlopak** [comment]
Also updated some packages.

File: src/utils/nextRunAt.ts [hunk 1]
```diff
@@ -28,21 +29,30 @@ export function isValidHumanInterval(value: unknown): value is string {
 export const computeFromInterval = (attrs: IJobParameters): Date => {
 	const previousNextRunAt = attrs.nextRunAt || new Date();
 	log('[%s:%s] computing next run via interval [%s]', attrs.name, attrs._id, attrs.repeatInterval);
+
 	const lastRun = dateForTimezone(attrs.lastRunAt || new Date(), attrs.repeatTimezone);
+
+	const cronOptions: ArgumentsType<typeof parseExpression>[1] = {
```
Comment (simllll) [inline src/utils/nextRunAt.ts:35]: no need for a custom "ArguementTypes"; just use the build in versino "Parameters".
https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype

File: src/utils/nextRunAt.ts [hunk 2]
```diff
@@ -28,21 +29,30 @@ export function isValidHumanInterval(value: unknown): value is string {
...
+		currentDate: lastRun.toDate()
+	};
+
+	if (attrs.repeatTimezone) {
```
Comment (simllll) [inline src/utils/nextRunAt.ts:39]: if timezone is not set, it should be therefore undefined, and setting cronOptions.tz = undefined shoul dbe equivalnt, if types allow this, I would remove this if, and set the timezone directly in like 36 (like currentDate)

File: src/types/ArgumentsType.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ArgumentsType<T extends (...args: any[]) => any> = T extends (...args: infer A) => any
```
Comment (simllll) [inline src/types/ArgumentsType.ts:2]: can be removed then

**Review by simllll — APPROVED** [review]

**Pull request merged**